irohad/multi_sig_transactions/gossip_propagation_strategy.hpp

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_GOSSIP_PROPAGATION_STRATEGY_HPP
+#define IROHA_GOSSIP_PROPAGATION_STRATEGY_HPP
+
+#include <chrono>
+#include <nonstd/optional.hpp>
+#include <rxcpp/rx.hpp>
+#include "ametsuchi/peer_query.hpp"
+#include "multi_sig_transactions/mst_propagation_strategy.hpp"
+
+namespace iroha {
+
+  /**
+   * This class provides strategy for propagation states in network
+   * Choose unique amount of peers each period of time
```

Comment:
It always chooses at most `amount` of peers, so the number is not always unique. Maybe `Choose at most n peers` sounds better.

Reply:
yep, thanks

Reply:
Good point, thanks